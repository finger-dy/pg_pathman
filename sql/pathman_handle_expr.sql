\set VERBOSITY terse

SET search_path = 'public';
CREATE SCHEMA pathman;
CREATE EXTENSION pg_pathman SCHEMA pathman;
CREATE SCHEMA test;

/* Init tables with range and hash partitioning */
CREATE TABLE test.range_partitioned(id integer not null, val real);
INSERT INTO test.range_partitioned SELECT generate_series(1, 10000), random();
SELECT pathman.create_range_partitions('test.range_partitioned', 'id', 1, 1000);
CREATE TABLE test.hash_partitioned(id integer not null, val real);
INSERT INTO test.hash_partitioned SELECT generate_series(1, 10000), random();
SELECT pathman.create_hash_partitions('test.hash_partitioned', 'id', 6);

/* Test select ... where id in (1, 2, 3, 4) */
EXPLAIN (COSTS OFF) SELECT * FROM test.range_partitioned WHERE id IN (1, 2, 3, 4);
EXPLAIN (COSTS OFF) SELECT * FROM test.hash_partitioned WHERE id IN (1, 2, 3, 4);

DROP SCHEMA test CASCADE;
DROP EXTENSION pg_pathman CASCADE;
DROP SCHEMA pathman CASCADE;
