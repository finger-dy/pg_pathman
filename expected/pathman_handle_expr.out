\set VERBOSITY terse
SET search_path = 'public';
CREATE SCHEMA pathman;
CREATE EXTENSION pg_pathman SCHEMA pathman;
CREATE SCHEMA test;
/* Init tables with range and hash partitioning */
CREATE TABLE test.range_partitioned(id integer not null, val real);
INSERT INTO test.range_partitioned SELECT generate_series(1, 10000), random();
SELECT pathman.create_range_partitions('test.range_partitioned', 'id', 1, 1000);
NOTICE:  sequence "range_partitioned_seq" does not exist, skipping
 create_range_partitions 
-------------------------
                      10
(1 row)

CREATE TABLE test.hash_partitioned(id integer not null, val real);
INSERT INTO test.hash_partitioned SELECT generate_series(1, 10000), random();
SELECT pathman.create_hash_partitions('test.hash_partitioned', 'id', 6);
 create_hash_partitions 
------------------------
                      6
(1 row)

/* Test select ... where id in (1, 2, 3, 4) */
EXPLAIN (COSTS OFF) SELECT * FROM test.range_partitioned WHERE id IN (1, 2, 3, 4);
                     QUERY PLAN                      
-----------------------------------------------------
 Append
   ->  Seq Scan on range_partitioned_1
         Filter: (id = ANY ('{1,2,3,4}'::integer[]))
(3 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test.hash_partitioned WHERE id IN (1, 2, 3, 4);
                     QUERY PLAN                      
-----------------------------------------------------
 Append
   ->  Seq Scan on hash_partitioned_2
         Filter: (id = ANY ('{1,2,3,4}'::integer[]))
   ->  Seq Scan on hash_partitioned_3
         Filter: (id = ANY ('{1,2,3,4}'::integer[]))
   ->  Seq Scan on hash_partitioned_4
         Filter: (id = ANY ('{1,2,3,4}'::integer[]))
   ->  Seq Scan on hash_partitioned_5
         Filter: (id = ANY ('{1,2,3,4}'::integer[]))
(9 rows)

DROP SCHEMA test CASCADE;
NOTICE:  drop cascades to 19 other objects
DROP EXTENSION pg_pathman CASCADE;
DROP SCHEMA pathman CASCADE;
